server/http/test_http/oauth2_complete_test.go

```diff
@@ -271,24 +313,83 @@ func statusOKGraphAPIResponderFunc() {
 	httpmock.RegisterResponder("GET", meRequestURL, meResponder)
 }
 
-func getMockBotPoster(ctrl *gomock.Controller, err error) utils.BotPoster {
+func getBotPosterMessage(displayName string) string {
+	return fmt.Sprintf("### Welcome to the Microsoft Office plugin!\n"+
+		"Here is some info to prove we got you logged in\n"+
+		"Name: %s \n", displayName)
+}
+
+type mockBotPosterConfig struct {
```

Comment:
Hmm, it's an interesting approach. But you probably realized that you're spending a lot of time handling the logic of the mocks, and then you'd have to add more logic if you want to do something else with the mocks. 

I probably would have gone with a `setupMocks(mock1, mock2, ...etc)` function that you could define for each test case, which would do the setups for the mocks for that particular case. That way you don't have to worry about config structs at all. The main test would call each test case's `setupMocks` function to prepare the mocks before making the call. Make sense?

Reply:
That makes sense. I didn't think about passing in the mocks to a function for setup. That would make it much more clean than a config and doing logical steps for different configurations.